op-program/host/prefetcher/prefetcher.go

```diff
@@ -28,6 +28,11 @@ var (
 	precompileFailure = [1]byte{0}
 )
 
+var (
+	ErrExperimentalPrefetchFailed   = errors.New("experimental prefetch failed")
+	ErrExperimentalPrefetchDisabled = errors.New("experimental prefetch disabled")
+)
```

Comment:
Can we define these in `l2_source` since that's where they're returned from?  Then the dependency is that prefetcher uses l2_source rather than having it go both ways as well.

Reply:
Looks like we import prefetcher from host, and l2_source is currently in the host package, so moving this causes an import cycle between host and prefetcher. Should I extract l2_source to a new package or move l2_source to client/l2?

Reply:
`l2_source` is in the `host` package, so if we move these to the host package, then we can't import these errors in the `prefetcher` package. Should I move one of these to a new package (l2_source could go in client/l2/source instead, or I could move the error to an errors package somewhere).

Reply:
hmm, let's leave them here for now then, but see if we can follow up and move l2_source into a better package - it doesn't belong under the `client` package as it's definitely part of the host side but not sure it deserves to be at the top level either.  Wherever it lands, probably better to avoid creating extra noise in these PRs.

Reply:
Created https://github.com/ethereum-optimism/optimism/issues/12677 to track this.